test/test_base_metric.py

```diff
@@ -5,86 +5,108 @@
 from sklearn import clone
 from sklearn.utils.testing import set_random_state
 from test.test_utils import ids_metric_learners, metric_learners
+import string
+
+
+def remove_spaces(s):
+  s = s.replace(' ', '')
+  s = s.replace('\n', '')
+  return s
 
 
 class TestStringRepr(unittest.TestCase):
 
   def test_covariance(self):
-    self.assertEqual(str(metric_learn.Covariance()),
-                     "Covariance(preprocessor=None)")
+    self.assertEqual(remove_spaces(str(metric_learn.Covariance())),
+                     remove_spaces("Covariance(preprocessor=None)"))
 
   def test_lmnn(self):
     self.assertRegexpMatches(
         str(metric_learn.LMNN()),
         r"(python_)?LMNN\(convergence_tol=0.001, k=3, learn_rate=1e-07, "
-        r"max_iter=1000,\n      min_iter=50, preprocessor=None, "
-        r"regularization=0.5, use_pca=True,\n      verbose=False\)")
+        r"max_iter=1000,\n(\ +)min_iter=50, preprocessor=None, "
```

Comment:
`\s+` is a bit more readable, IMO

Reply:
I agree, it's better, I didn't know `\s`